src/platform/native.zig

```diff
@@ -587,60 +587,81 @@ pub const Platform = struct {
 
 pub const BackingTimer = std.time.Timer;
 
+var app: App = undefined;
+
 pub fn main() !void {
     var gpa = std.heap.GeneralPurposeAllocator(.{}){};
     defer _ = gpa.deinit();
     const allocator = gpa.allocator();
 
-    var core = try Core.init(allocator);
-    defer core.internal.deinit();
-    var app: App = undefined;
-
-    try app.init(&core);
-    defer app.deinit(&core);
-
-    // Glfw specific: initialize the user pointer used in callbacks
-    core.internal.initCallback();
-
-    const window = core.internal.window;
-    while (!window.shouldClose()) {
-        if (core.internal.wait_event_timeout > 0.0) {
-            if (core.internal.wait_event_timeout == std.math.inf(f64)) {
-                // Wait for an event
-                try glfw.waitEvents();
-            } else {
-                // Wait for an event with a timeout
-                try glfw.waitEventsTimeout(core.internal.wait_event_timeout);
-            }
+    var core = try core_init(allocator);
+    defer core_deinit(core);
+
+    try app.init(core);
+    defer app.deinit(core);
+
+    while (!core.internal.window.shouldClose()) {
+        try core_update(core, null);
+
+        try app.update(core);
+    }
+}
+
+pub fn core_init(allocator: std.mem.Allocator) !*Core {
+    const core: *Core = try allocator.create(Core);
+    core.* = try Core.init(allocator);
+
+    // // Glfw specific: initialize the user pointer used in callbacks
+    core.*.internal.initCallback();
+
+    return core;
+}
+
+pub export fn core_deinit(core: *Core) void {
```

Comment:
I think this function shouldn't be exported.

Also, while `snake_case` is correct in C (and `libmach.zig`), here it is not. The function names in this file should be `coreDeinit`, `coreInit`, `coreUpdate` - quoting the [Zig style guide](https://ziglang.org/documentation/master/#toc-Names):

> Roughly speaking: camelCaseFunctionName, TitleCaseTypeName, snake_case_variable_name. More precisely: 

Of course what you did in `libmach.zig` is correct, since those names must match C conventions as exported functions.